### PR TITLE
test: speed up unit tests by using smaller datasets

### PR DESCRIPTION
**Pull request by fengelniederhammer**
part of #560 

### Summary

Local results (debug build):
* database.test.cpp: Use a smaller dataset. 7s -> 800ms
* preprocessor.test.cpp: The RSV dataset took 45s -> shift it to the new e2e tests and delete here
* preprocessor.test.cpp: Use smaller datasets for some tests: 1.8s -> ~500ms

On CI, time of the unit tests:
* last run on main: 16s
* first build on this branch: 2s

## PR Checklist

~~- [ ] All necessary documentation has been adapted or there is an issue to do so.~~
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: testBaseData/unitTestDummyDataset/metadata.tsv_ [hunk 1]
```diff
@@ -0,0 +1,6 @@
+primaryKey	pango_lineage	date	region	country	division	unsorted_date	age	qc_value	test_boolean_column
```
Comment (Taepper) [inline testBaseData/unitTestDummyDataset/metadata.tsv_:1]: Is this file added on purpose?

Comment (fengelniederhammer) [reply testBaseData/unitTestDummyDataset/metadata.tsv_:1]: No, thanks...

**Review by Taepper — APPROVED** [review]

**Comment by Taepper** [comment]
I added https://github.com/GenSpectrum/LAPIS-SILO-e2e/issues/34 to not drop our covered input data

**Comment by fengelniederhammer** [comment]
> I added [GenSpectrum/LAPIS-SILO-e2e#34](https://github.com/GenSpectrum/LAPIS-SILO-e2e/issues/34) to not drop our covered input data

IMO the dataset is not of immediate importance. In these tests here, it was just some dataset. We already have some other dataset in other tests.
In the longterm of course still good to have.

**Comment by Taepper** [comment]
Exactly. I just wanted to have the issues such that we remember adding it at some point. Not blocking in any way IMO :)

**Review by pflanze — APPROVED** [review]
Cool. I haven't looked into all of the details, but looks good to me. 

(It's good to have a test suite that covers most of the potential issues when coding, and then have other more expensive tests/test suites that give more certainty for a higher cost. Even in unit tests, could have expensive ones if there are 2 classes of tests, important-but-quick and more-extensive-but-expensive ones.)

**Comment by Taepper** [comment]
We could also learn from other codebases. e.g. duckdb marks suffixes some test-files with .test_slow:
https://github.com/duckdb/duckdb/tree/main/test/sql/copy

**Comment by Taepper** [comment]
And then their test runner can execute them based on the arguments given by the developer running the tests

**Pull request merged**